### PR TITLE
Tests for mapreduce on empty collection

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/reduce.jl [hunk 1]
```diff
@@ -44,6 +44,13 @@
 @test mapreduce(abs2, +, [-9, -3]) == 81 + 9
 @test mapreduce(-, +, [-9, -3, -4, 8, -2]) == (9 + 3 + 4 - 8 + 2)
 @test mapreduce(-, +, collect(linspace(1.0, 10000.0, 10000))) == -50005000.0
+# empty mr
+@test mapreduce(abs2, +, Float64[]) == 0
```
Comment (nalimilan) [inline test/reduce.jl]: Wouldn't it make sense to also check the type using `=== 0.0`?

**Review by nalimilan — APPROVED** [review]
Thanks!

**Comment by kshyatt** [comment]
Looks like a build timeout on linux. Are we good to merge?

**Comment by nalimilan** [comment]
Yes, that was a timeout, I've restarted the build just in case.

**Pull request merged**